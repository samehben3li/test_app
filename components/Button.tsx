import { Pressable, Image, Text } from "react-native";

const Button = ({
  fn,
  icon,
  style,
  route,
  image,
  name,
  variant,
  globalStyles,
}: {
  fn: () => void;
  icon?: any;
  style: any;
  variant: string;
  route?: any;
  image?: any;
  name?: string;
  globalStyles: any;
}) => {
  return variant === "home" ? (
    <Pressable style={globalStyles.btn} onPress={fn}>
      <Image style={style} source={icon} />
      <Text style={globalStyles.btnText}>NEW FLAG</Text>
    </Pressable>
  ) : (
    <Pressable style={style} onPress={fn}>
      <Image
        style={[
          globalStyles.image,
          route.name === name && globalStyles.selected,
        ]}
        source={image}
      />
    </Pressable>
  );
};

export default Button;
