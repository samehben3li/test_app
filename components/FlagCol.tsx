import { Text, View, Image } from "react-native";
import { flagListStyles as styles } from "../styles";
import i18n from "../i18n/translations";
import { flagTab } from "../types/interfaces";

const API_URI = process.env.apiUrl;

interface Props {
  name: string;
  item: flagTab;
}

const FlagCol = ({ item, name }: Props) => (
  <View style={styles.col}>
    <View style={styles.title}>
      <Text style={styles.titleTxt}>{i18n.t(name)}</Text>
    </View>
    <View style={styles.selection}>
      <Image
        style={styles.image}
        source={{ uri: `${API_URI}${item?.imgUrl}` }}
      />
    </View>
    <View style={styles.name}>
      <Text style={styles.nameTxt}>{item?.name}</Text>
    </View>
  </View>
);

export default FlagCol;
