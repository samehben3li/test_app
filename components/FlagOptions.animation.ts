import { Keyframe } from "react-native-reanimated";

const keyframe = new Keyframe({
  0: {
    opacity: 0,
    transform: [{ translateY: 10 }],
  },
  100: {
    opacity: 1,
    transform: [{ translateY: 0 }],
  },
});
export default keyframe;
