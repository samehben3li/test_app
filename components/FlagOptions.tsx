import { Text, View, Image, Pressable } from "react-native";
import { useState, useEffect } from "react";
import i18n from "../i18n/translations";
import { flagOptionsStyles as styles } from "../styles";
import { selectedTab, flag, option } from "../screens/CreateFlag";
import GestureRecognizer from "react-native-swipe-gestures";
import Animated, { SlideInDown, SlideOutDown } from "react-native-reanimated";
import { API_URI } from "../utils/config";

interface Props {
  data: selectedTab;
  setFlagData: React.Dispatch<React.SetStateAction<flag>>;
  flagData: flag;
  setCompleted: React.Dispatch<React.SetStateAction<boolean>>;
  options: option[];
}

export default function FlagOptions({
  data,
  setFlagData,
  flagData,
  setCompleted,
  options,
}: Props) {
  const [selected, setSelected] = useState(0);
  useEffect(() => {
    if (flagData[data.name]) {
      setSelected(flagData[data.name].id);
    }
  }, [data]);

  const choose = (selection: option) => {
    setFlagData((prev) => {
      return { ...prev, [data.name]: selection };
    });
    setSelected(selection.id);
  };
  const addLocation = (location: string, side: string) => {
    let arr = flagData.location[side];
    const index = arr.indexOf(location);
    if (index > -1) {
      arr.splice(index, 1);
    } else {
      arr = [...arr, location];
    }
    const newValue = { ...flagData.location, [side]: arr };
    setFlagData((prev) => {
      return { ...prev, location: newValue };
    });
  };
  const locations = [
    i18n.t("flag.top"),
    i18n.t("flag.middle"),
    i18n.t("flag.bottom"),
  ];
  return (
    <Animated.View
      entering={SlideInDown}
      exiting={SlideOutDown}
      style={styles.container}
    >
      <GestureRecognizer
        onSwipeDown={() => {
          const { risk, plantPart, location, pest } = flagData;
          if (risk && plantPart && location && pest) {
            setCompleted(true);
          }
        }}
        style={styles.innerContainer}
      >
        <Text style={styles.hint}>{data.hint}</Text>
        <View style={styles.optionsContainer}>
          <Text style={styles.title}>{data.title}</Text>
          <View style={styles.options}>
            {!data.location ? (
              options?.map((option) => (
                <View key={option.id} style={styles.optionContainer}>
                  <Pressable
                    onPress={() => choose(option)}
                    style={[
                      styles.btn,
                      selected === option.id && styles.selected,
                    ]}
                  >
                    <Image
                      source={{ uri: `${API_URI}${option.imgUrl}` }}
                      style={styles.image}
                    />
                  </Pressable>
                  <Text style={styles.optionName}>{option.name}</Text>
                </View>
              ))
            ) : (
              <View style={styles.locationGrid}>
                <View style={styles.locationCol}>
                  {locations.map((item: string, index: number) => (
                    <Pressable
                      onPress={() => addLocation(item, "left")}
                      key={index}
                      style={[
                        styles.locationBtn,
                        flagData.location.left.includes(item) &&
                          styles.selected,
                      ]}
                    >
                      <Text style={styles.optionName}>{item}</Text>
                    </Pressable>
                  ))}
                  <Text style={styles.gridText}>{i18n.t("flag.left")}</Text>
                </View>
                <View style={styles.locationCol}>
                  {locations.map((item: string, index: number) => (
                    <Pressable
                      onPress={() => addLocation(item, "right")}
                      key={index}
                      style={[
                        styles.locationBtn,
                        flagData.location.right.includes(item) &&
                          styles.selected,
                      ]}
                    >
                      <Text style={styles.optionName}>{item}</Text>
                    </Pressable>
                  ))}
                  <Text style={styles.gridText}>{i18n.t("flag.right")}</Text>
                </View>
              </View>
            )}
          </View>
        </View>
      </GestureRecognizer>
    </Animated.View>
  );
}
