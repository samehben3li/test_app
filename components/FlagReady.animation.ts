/* istanbul ignore file */
import { Keyframe } from "react-native-reanimated";
/* istanbul ignore next */
const keyframe2 = new Keyframe({
  0: {
    width: 0,
    height: 0,
  },
  100: {
    height: 50,
    width: 150,
  },
});

export default keyframe2;
