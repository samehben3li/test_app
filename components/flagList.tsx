import { Text, View, ScrollView, Image } from "react-native";
import i18n from "../i18n/tanslations";
import styles from "../styles/flagList.style";
import flagIcon from "../assets/flag.png";
import bugIcon from "../assets/bug.png";
import { list } from "../screens/options";

export default function FlagList() {
  return (
    <ScrollView style={styles.container}>
      {list.map((item, index) => (
        <View key={index} style={styles.flagContainer}>
          <View style={styles.header}>
            <Image source={flagIcon} style={styles.img} />
            <Text style={styles.date}>{item.date}</Text>
          </View>
          <View style={styles.grid}>
            <View style={styles.col}>
              <View style={styles.title}>
                <Text style={styles.titleTxt}>{i18n.t("riskCategory")}</Text>
              </View>
              <View style={styles.selection}>
                <Image style={styles.image} source={bugIcon} />
              </View>
              <View style={styles.name}>
                <Text style={styles.nameTxt}>{item.risk}</Text>
              </View>
            </View>
            <View style={styles.col}>
              <View style={styles.title}>
                <Text style={styles.titleTxt}>{i18n.t("pestType")}</Text>
              </View>
              <View style={styles.selection}>
                <Image style={styles.image} source={bugIcon} />
              </View>
              <View style={styles.name}>
                <Text style={styles.nameTxt}>{item.type}</Text>
              </View>
            </View>
            <View style={styles.col}>
              <View style={styles.title}>
                <Text style={styles.titleTxt}>{i18n.t("plantPart")}</Text>
              </View>
              <View style={styles.selection}>
                <Image style={styles.image} source={bugIcon} />
              </View>
              <View style={styles.name}>
                <Text style={styles.nameTxt}>{item.part}</Text>
              </View>
            </View>
            <View style={styles.col}>
              <View style={styles.title}>
                <Text style={styles.titleTxt}>{i18n.t("location")}</Text>
              </View>
              <View style={styles.selection}>
                <Image style={styles.image} source={bugIcon} />
              </View>
              <View style={styles.name}>
                <Text style={styles.nameTxt}>{item.location}</Text>
              </View>
            </View>
          </View>
        </View>
      ))}
    </ScrollView>
  );
}
