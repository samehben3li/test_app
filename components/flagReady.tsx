import { Text, View, Image, Pressable } from "react-native";
import { useState, useEffect } from "react";
import i18n from "../i18n/tanslations";
import styles from "../styles/flagReady.style";
import helpIcon from "../assets/help.png";
import lunaHelperIcon from "../assets/lunaHelper.png";
import GestureRecognizer from "react-native-swipe-gestures";
import { selectedTab, flag, option } from "../screens/createFlag";
import { CREATE_FLAG } from "../requests/mutations";
import { useMutation } from "@apollo/client";

interface Props {
  setCompleted: React.Dispatch<React.SetStateAction<boolean>>;
  flagData: flag;
}

export default function FlagReady({ setCompleted, flagData }: Props) {
  const [createFlag] = useMutation(CREATE_FLAG);
  const [loading, setLoading] = useState(false);
  const [done, setDone] = useState(false);
  const submit = async () => {
    setLoading(true);
    const { risk, plantPart, location, pest } = flagData;
    const inputValues = {
      riskCategory: {
        name: risk?.name,
        imgUrl: risk?.imgUrl,
      },
      riskCategoryType: {
        name: pest?.name,
        imgUrl: pest?.imgUrl,
      },
      plantPart: {
        name: plantPart?.name,
        imgUrl: plantPart?.imgUrl,
      },
      location: location,
    };
    console.log(inputValues);
    try {
      const data = await createFlag({ variables: inputValues });
      if (data) {
        setLoading(false);
        setDone(true);
      }
    } catch (err) {
      if (err instanceof Error) {
        setLoading(false);
        alert(err.message);
      }
    }
  };
  useEffect(() => {
    let timer: any;
    if (done) {
      timer = setTimeout(() => setDone(false), 5000);
    }
    return () => {
      clearTimeout(timer);
    };
  }, [done]);
  return (
    <View style={styles.container}>
      {done ? (
        <Text>done</Text>
      ) : (
        <>
          <View style={styles.topPart}>
            <View style={styles.triangle}></View>
            <View style={styles.helpContainer}>
              <View style={styles.header}>
                <Image source={lunaHelperIcon} style={styles.imgWide} />
                <Image source={helpIcon} style={styles.imgIcon} />
              </View>
              <Text style={styles.helpText}>{i18n.t("makeChangesHelp")}</Text>
            </View>
          </View>
          <GestureRecognizer
            onSwipeDown={() => setCompleted(false)}
            style={styles.bottomPart}
          >
            <Text style={styles.confirmText}>{i18n.t("confirmSelection")}</Text>
            <Pressable
              onPress={submit}
              style={[
                styles.btn,
                {
                  opacity: !loading ? 1 : 0.5,
                },
              ]}
            >
              <Text style={styles.btnText}>{i18n.t("createFlag")}</Text>
            </Pressable>
            <Text style={styles.hint}>{i18n.t("swipeHint")}</Text>
          </GestureRecognizer>
        </>
      )}
    </View>
  );
}
