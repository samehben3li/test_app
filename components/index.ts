export { default as Header } from "./Header";
export { default as FlagList } from "./FlagList";
export { default as FlagOptions } from "./FlagOptions";
export { default as FlagReady } from "./FlagReady";
export { default as NewFlag } from "./NewFlag";
export { default as ScreenSwitch } from "./ScreenSwitch";
export { default as FlagCol } from "./FlagCol";
