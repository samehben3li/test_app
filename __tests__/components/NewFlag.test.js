import React from "react";
import renderer from "react-test-renderer";
import { NewFlag } from "../../components";
import { MockedProvider } from "@apollo/client/testing";
import AuthProvider from "../../context/authContext";

describe("NewFlag Component", () => {
  it("renders without crashing", () => {
    const rendered = renderer
      .create(
        <MockedProvider>
          <AuthProvider value={{}}>
            <NewFlag
              selectedTab={{
                name: "selectedTabName",
                title: "selectedTilte",
                hint: "selectedHint",
                options: [],
                location: false,
              }}
              flagData={{
                risk: null,
                pest: null,
                plantPart: null,
                location: {
                  left: ["TOP"],
                  right: ["BOTTOM"],
                },
              }}
            />
          </AuthProvider>
        </MockedProvider>
      )
      .toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
