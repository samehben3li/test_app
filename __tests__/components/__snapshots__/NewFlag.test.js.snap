// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewFlag Component renders without crashing 1`] = `
<View
  animatedStyle={
    Object {
      "value": Object {},
    }
  }
  collapsable={false}
  entering={
    Keyframe {
      "build": [Function],
      "definitions": Object {
        "0": Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateY": -100,
            },
          ],
        },
        "100": Object {
          "easing": [Function],
          "opacity": 1,
          "transform": Array [
            Object {
              "translateY": 0,
            },
          ],
        },
      },
      "durationV": 600,
    }
  }
  exiting={
    SlideOutUp {
      "build": [Function],
      "durationV": 600,
      "randomizeDelay": false,
    }
  }
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#fff",
      "borderRadius": 15,
      "elevation": 24,
      "flex": 1,
      "justifyContent": "center",
      "margin": 20,
      "marginBottom": 40,
      "padding": 15,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 12,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 16,
      "zIndex": 9,
    }
  }
>
  <Image
    animatedStyle={
      Object {
        "value": Object {},
      }
    }
    collapsable={false}
    entering={
      FadeIn {
        "build": [Function],
        "delayV": 700,
        "durationV": 700,
        "randomizeDelay": false,
      }
    }
    source={1}
    style={
      Object {
        "height": 25,
        "marginBottom": 13,
        "marginLeft": 5,
        "resizeMode": "contain",
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      animatedStyle={
        Object {
          "value": Object {},
        }
      }
      collapsable={false}
      entering={
        FadeIn {
          "build": [Function],
          "delayV": 300,
          "randomizeDelay": false,
        }
      }
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "elevation": 20,
          "flex": 1,
          "marginHorizontal": 4,
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 13.16,
          "zIndex": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#cccccc",
            "borderBottomWidth": 2,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#cccccc",
                "fontFamily": "bold",
                "fontSize": 10,
                "lineHeight": 15,
                "textAlign": "center",
                "width": "90%",
              },
              null,
            ]
          }
        >
          RISK CATEGORY
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "justifyContent": "center",
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#cccccc",
              "borderBottomEndRadius": 5,
              "borderBottomStartRadius": 5,
              "flex": 1,
              "justifyContent": "center",
            },
            null,
            undefined,
            null,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "black",
                "fontSize": 15,
              },
              null,
            ]
          }
        >
          ?
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      animatedStyle={
        Object {
          "value": Object {},
        }
      }
      collapsable={false}
      entering={
        FadeIn {
          "build": [Function],
          "delayV": 300,
          "randomizeDelay": false,
        }
      }
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "elevation": 20,
          "flex": 1,
          "marginHorizontal": 4,
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 13.16,
          "zIndex": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#cccccc",
            "borderBottomWidth": 2,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#cccccc",
                "fontFamily": "bold",
                "fontSize": 10,
                "lineHeight": 15,
                "textAlign": "center",
                "width": "90%",
              },
              null,
            ]
          }
        >
          TYPE
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "justifyContent": "center",
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#cccccc",
              "borderBottomEndRadius": 5,
              "borderBottomStartRadius": 5,
              "flex": 1,
              "justifyContent": "center",
            },
            null,
            undefined,
            null,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "black",
                "fontSize": 15,
              },
              null,
            ]
          }
        >
          ?
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      animatedStyle={
        Object {
          "value": Object {},
        }
      }
      collapsable={false}
      entering={
        FadeIn {
          "build": [Function],
          "delayV": 300,
          "randomizeDelay": false,
        }
      }
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "elevation": 20,
          "flex": 1,
          "marginHorizontal": 4,
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 13.16,
          "zIndex": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#cccccc",
            "borderBottomWidth": 2,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#cccccc",
                "fontFamily": "bold",
                "fontSize": 10,
                "lineHeight": 15,
                "textAlign": "center",
                "width": "90%",
              },
              null,
            ]
          }
        >
          PLANT PART
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "justifyContent": "center",
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#cccccc",
              "borderBottomEndRadius": 5,
              "borderBottomStartRadius": 5,
              "flex": 1,
              "justifyContent": "center",
            },
            null,
            undefined,
            null,
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "black",
                "fontSize": 15,
              },
              null,
            ]
          }
        >
          ?
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      animatedStyle={
        Object {
          "value": Object {},
        }
      }
      collapsable={false}
      entering={
        FadeIn {
          "build": [Function],
          "delayV": 300,
          "randomizeDelay": false,
        }
      }
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fff",
          "borderRadius": 5,
          "elevation": 20,
          "flex": 1,
          "marginHorizontal": 4,
          "opacity": 1,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 10,
            "width": 0,
          },
          "shadowOpacity": 0.3,
          "shadowRadius": 13.16,
          "zIndex": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "borderBottomColor": "#cccccc",
            "borderBottomWidth": 2,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#cccccc",
                "fontFamily": "bold",
                "fontSize": 10,
                "lineHeight": 15,
                "textAlign": "center",
                "width": "90%",
              },
              false,
            ]
          }
        >
          LOCATION
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 2,
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "height": "80%",
                "opacity": 0.4,
                "overflow": "hidden",
                "width": "70%",
              },
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                  },
                  Object {
                    "backgroundColor": "#FFB955",
                  },
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                  },
                  false,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                  },
                  false,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                  },
                  false,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                  },
                  false,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "borderWidth": 1,
                    "flex": 1,
                  },
                  Object {
                    "backgroundColor": "#FFB955",
                  },
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#cccccc",
              "borderBottomEndRadius": 5,
              "borderBottomStartRadius": 5,
              "flex": 1,
              "justifyContent": "center",
            },
            false,
            Object {
              "backgroundColor": "#5CE166",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "black",
                "fontSize": 10,
                "textAlign": "center",
              },
              Object {
                "color": "#000",
                "fontSize": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          LEFT: 
          T
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "black",
                "fontSize": 10,
                "textAlign": "center",
              },
              Object {
                "color": "#000",
                "fontSize": 10,
                "textAlign": "center",
              },
            ]
          }
        >
          RIGHT: 
          B
        </Text>
      </View>
    </View>
  </View>
</View>
`;
